### PR TITLE
NMS-15693: XML Doc updates

### PR DESCRIPTION
**Pull request by mmahacek**
### All Contributors

* [X] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [X] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### External References

* Jira (Issue Tracker): https://opennms.atlassian.net/browse/NMS-15693

### CODE REVIEW

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 1]
```diff
@@ -79,16 +70,40 @@ This snippet provides a collection definition named `xml-opennms-nodes`:
...
+[source, xml]
+----
+<xml-groups>
+    <xml-group name="nodes" resource-type="node" resource-xpath="/nodes">
+        <xml-object name="totalCount" type="GAUGE" xpath="@totalCount"/>
+    </xml-group>
+</xml-groups>
+----
+
+=== Customizing the source request
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:91]:
```suggestion
=== Customize the source request
```

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 2]
```diff
@@ -141,13 +149,58 @@ If the endpoint being collected requires additional headers, you can add them as
   </xml-source>
 ----
 
-== Test XML collection via Karaf
+=== Retrieving data from a POST request
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:152]:
```suggestion
=== Retrieve data from a POST request
```

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 3]
```diff
@@ -65,10 +57,9 @@ If the root element is an array, it will be wrapped in an object accessible as `
 
 Understanding <<operation:deep-dive/performance-data-collection/resource-types.adoc#resource-types, resource types>> helps when editing collector-specific configuration files.
 
-XML Collections are defined in `etc/xml-datacollection-config.xml` and `etc/xml-datacollection/`.
-
-This snippet provides a collection definition named `xml-opennms-nodes`:
+XML Collections are defined in `$\{OPENNMS_HOME}/etc/xml-datacollection-config.xml` and groups are defined in `$\{OPENNMS_HOME}/etc/xml-datacollection/`.
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:60]:
```suggestion
XML collections are defined in `$\{OPENNMS_HOME}/etc/xml-datacollection-config.xml` and groups are defined in `$\{OPENNMS_HOME}/etc/xml-datacollection/`.
```

**Review by Bonrob2 — CHANGES_REQUESTED** [review]
Minor changes. Thanks!

**Review by Bonrob2 — DISMISSED** [review]
LGTM! Thank you!

File: docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc [hunk 4]
```diff
@@ -113,20 +128,13 @@ The request element can have the following optional child elements:
...
-[source, xml]
-----
-<xml-groups>
-    <xml-group name="nodes" resource-type="node" resource-xpath="/nodes">
-        <xml-object name="totalCount" type="GAUGE" xpath="@totalCount"/>
-    </xml-group>
-</xml-groups>
-----
+| xslt-source-file
+| Full path to XSLT file to transform XML data before processing to collection definitions
```
Comment (Bonrob2) [inline docs/modules/reference/pages/performance-data-collection/collectors/xml.adoc:133]:
```suggestion
| Full path to XSLT file to transform XML data before processing to collection definitions.
```

**Review by agalue — APPROVED** [review]
LGTM

**Review by Bonrob2 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**